File: jcrm-client-web/src/main/java/con/ss/jcrm/client/web/resource/ContactMessengerResource.java
package con.ss.jcrm.client.web.resource;

import com.ss.jcrm.client.api.ContactMessenger;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ContactMessengerResource {

    private String login;
    private String type;

    public ContactMessengerResource(@NotNull ContactMessenger messenger) {
        this.login = messenger.getLogin();
        this.type = messenger.getType().name();
    }
}

File: jcrm-ui/src/app/resources/contact-messenger-resource.ts
import {ContactMessenger} from '@app/entity/contact-messenger';

export class ContactMessengerResource {

    public static valueOf(messenger: ContactMessenger): ContactMessengerResource {
        return new ContactMessengerResource(messenger.login, messenger.type);
    }

    constructor(
        public login: string,
        public type: string,
    ) {
    }
}
